docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_data_events_remediation.mdx

```diff
@@ -1,6 +1,225 @@
+   - If you want to modify an existing trail, select the trail from the list and click on its name to open its configuration page.
+
+3. **Enable Data Events:**
+   - In the trail configuration page, scroll down to the "Data events" section.
+   - Click on "Add data event selector."
+   - Choose the data event types you want to log (e.g., S3, Lambda).
+   - Specify the resources (e.g., all S3 buckets or specific Lambda functions) for which you want to log data events.
+
+4. **Save Changes:**
+   - After configuring the data events, click on "Create" if you are creating a new trail or "Save changes" if you are modifying an existing trail.
+
+By following these steps, you ensure that data events are logged in CloudTrail, which helps in monitoring and auditing activities in your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that AWS CloudTrail logs data events, you need to configure your CloudTrail to include data events for the specified resources. Here are the steps to achieve this using the AWS CLI:
+
+1. **Create or Update a CloudTrail Trail:**
+   If you don't have a CloudTrail trail, you need to create one. If you already have a trail, you can update it to include data events.
+
+   ```sh
+   aws cloudtrail create-trail --name my-trail --s3-bucket-name my-bucket
+   ```
+
+2. **Add Data Events to the Trail:**
+   Use the `put-event-selectors` command to specify the data events you want to log. For example, to log S3 object-level API operations:
+
+   ```sh
+   aws cloudtrail put-event-selectors --trail-name my-trail --event-selectors '[{"ReadWriteType": "All", "IncludeManagementEvents": true, "DataResources": [{"Type": "AWS::S3::Object", "Values": ["arn:aws:s3:::"]}]}]'
+   ```
+
+3. **Verify the Configuration:**
+   Ensure that the event selectors have been correctly applied to your trail by describing the trail:
+
+   ```sh
+   aws cloudtrail get-event-selectors --trail-name my-trail
+   ```
+
+4. **Enable the Trail:**
+   If the trail is not already enabled, you need to start logging:
+
+   ```sh
+   aws cloudtrail start-logging --name my-trail
+   ```
+
+By following these steps, you can ensure that CloudTrail logs data events, which is crucial for comprehensive monitoring and auditing of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that CloudTrail logs data events in AWS, you can use the Boto3 library in Python. Here are the steps to prevent the misconfiguration where CloudTrail is not logging data events:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with the necessary AWS credentials and region.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>